Prefix local macros for improved usability with Edition 2018

abreis opened the pull request:
This PR prefixes the `$crate` variable to all invocations of local macros in this library.

With Rust 2018, the recommended approach is now to `use macro` instead of `extern crate`. With this PR, the user is able to invoke `delegate!` with a single import:
```rust
use delegate::delegate;
```
instead of having to bring every local macro into view:
```rust
use delegate::{delegate, delegate__ensure_self, delegate__expand, delegate__parse};
```
@chancancode would you like to see the `README.md` updated as well?

Closes #6.

abreis commented:
Okay, so it seems I have a bug here. I could not find a way to add the `$crate` modifier to the assignment of `delegate__expand` to `action` in [L14](https://github.com/chancancode/rust-delegate/pull/7/files#diff-b4aea3e418ccdb71239b96952d9cddb6R14):
```rust
action: delegate__expand
```
so I added it to the macro expansion in [L41](https://github.com/chancancode/rust-delegate/pull/7/files#diff-b4aea3e418ccdb71239b96952d9cddb6R41) instead:
```rust
$crate::$action ! { $($items)* }
```
Which obviously doesn't work if `action` is not local to the crate.

Not sure how to fix this, can you help?

abreis commented:
Solved, just had to switch `$action` from a token tree to a path.

chancancode commented:
Wheee thank you!!! @abreis yes, I think we should update the README. AFAICT this is not otherwise a break changing right? (i.e. the old style "import" will still work?)

abreis commented:
Correct, `#[macro_use]` is not deprecated and will simply pull all of the crate's macros into scope.

I've updated the Readme to the new style and also fixed a few typos. Should be good to go!

chancancode commented:
@abreis sorry it took so long! just catching up from the holidays 😓 published as v0.2.0 too
